File: csharp/demo/demo04-sleep.cs
/*
 * SLEEP
 */
using System;
using System.Threading;



class Demo04 : IRunnable
{
    public void run()
    {
        var thFoo = new Thread(() =>
        {
            Console.WriteLine("foo is sleeping");
            Thread.Sleep(3000);
            Console.WriteLine("foo wakes up");
        });

        thFoo.Start();
        thFoo.Join();

        Console.WriteLine("Good bye");
    }
}

File: python/demo04_sleep.py
'''
SLEEP
'''

import time
import threading



def do_task():
    print('foo is sleeping')
    time.sleep(3)
    print('foo wakes up')



th_foo = threading.Thread(target=do_task)

th_foo.start()
th_foo.join()

print('Good bye')
